aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -237,15 +303,66 @@ module aptos_framework::coin {
         coin_store.frozen = true
     }
 
+    /// Upgrade total supply to use a parallelizable implementation if it is
+    /// available.
+    public entry fun upgrade_supply<CoinType>(account: &signer) acquires CoinInfo, SupplyConfig {
+        let account_addr = signer::address_of(account);
+
+        // Only coin creators can upgrade supply.
+        assert!(
+            coin_address<CoinType>() == account_addr,
+            error::invalid_argument(ECOIN_INFO_ADDRESS_MISMATCH),
+        );
+
+        // Can only succeed once on-chain governance agreed on the upgrade.
+        assert!(
+            borrow_global_mut<SupplyConfig>(@aptos_framework).allow_upgrades,
+            error::permission_denied(ECOIN_SUPPLY_UPGRADE_NOT_SUPPORTED)
+        );
+
+        let supply = &mut borrow_global_mut<CoinInfo<CoinType>>(account_addr).supply;
+        if (option::is_some(supply)) {
+            let supply = option::borrow_mut(supply);
+
+            // If supply is tracked and the current implementation uses an integer - upgrade.
+            if (!optional_aggregator::is_parallelizable(&supply.inner)) {
+                optional_aggregator::switch(&mut supply.inner);
+            }
+        }
+    }
+
     /// Creates a new Coin with given `CoinType` and returns minting/burning capabilities.
-    /// The given signer also becomes the account hosting the information
-    /// about the coin (name, supply, etc.).
+    /// The given signer also becomes the account hosting the information  about the coin
+    /// (name, supply, etc.). Supply is initialized as non-parallelizable integer.
     public fun initialize<CoinType>(
         account: &signer,
         name: string::String,
         symbol: string::String,
         decimals: u8,
         monitor_supply: bool,
+    ): (BurnCapability<CoinType>, FreezeCapability<CoinType>, MintCapability<CoinType>) {
+        initialize_internal(account, name, symbol, decimals, monitor_supply, false)
+    }
+
+    /// Same as `initialize` but supply can be initialized to parallelizable aggregator.
+    public fun initialize_with_parallelizable_supply<CoinType>(
```

Comment:
This is only meant to be callable in/for AptosCoin right? If so, this should be friend only.

Reply:
@movekevin any coin on aptos account, since it checks signer:
```
system_addresses::assert_aptos_framework(account);
```

Reply:
Do we want Aptos coin only?